src/codegen/templates/property_setter_template.py.mako

```diff
@@ -43,15 +44,22 @@
             function_call_args.append("\"\"")
         function_call_args.append(hex(attribute.id))
         function_call_args.append('val')
-        if attribute.name in ACTIVE_DEVS_SUPPORTED_ATTRIBUTES:
+        if attribute.name in active_devs_supported_attributes:
             function_call_args_ex = function_call_args.copy()
             function_call_args_ex.insert(1, "self._active_devs")
     %>\
-    %if attribute.name in ACTIVE_DEVS_SUPPORTED_ATTRIBUTES:
+    %if attribute.python_class_name == "Timing":
+        %if attribute.name in active_devs_supported_attributes:
         if self._active_devs:
             self._interpreter.set_${generic_attribute_func_ex}(${', '.join(function_call_args_ex)})
         else:
             self._interpreter.set_${generic_attribute_func}(${', '.join(function_call_args)})
+        %else:
+        if self._active_devs == None:
+            self._interpreter.set_${generic_attribute_func}(${', '.join(function_call_args)})
+        else:
+            self._raise_device_context_not_supported_error()
```

Comment:
[style] The balanced if/else looks weird to me since one case is far less likely than the other.

I think this looks better:
```
    @change_detect_di_falling_edge_physical_chans.setter
    def change_detect_di_falling_edge_physical_chans(self, val):
        val = val.name
        if self._active_devs:
            self._raise_device_context_not_supported_error()
        self._interpreter.set_timing_attribute_string(self._handle, 0x2196, val)
```